server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -727,6 +729,10 @@ private void sync(final Consumer<IndexShard> sync, final String source) {
         }
     }
 
+    public void updateIndexAnalyzers(IndexSettings indexSettings) throws IOException {
+        indexAnalyzers = analysisRegistry.build(indexSettings);
+    }
```

Comment:
It's not valid to adjust these inside any action (or from outside at all) as this would only adjust the instances available at the node were the actions runs. The correct pattern is to adjust the cluster state only on the master node and apply any changes due to metadata listeners. See my (a bit hacky) WIP https://github.com/crate/crate/commit/44f30eabefabc4ecf7a56d41682f7071b414e93a.